src/promptflow/promptflow/executor/_tool_resolver.py

```diff
@@ -107,7 +110,79 @@ def _convert_to_assistant_definition(self, assistant_definition_path: str, input
         file = self._working_dir / assistant_definition_path
         with open(file, "r", encoding="utf-8") as file:
             assistant_definition = load_yaml(file)
-        return AssistantDefinition.deserialize(assistant_definition)
+        assistant_def = AssistantDefinition.deserialize(assistant_definition)
+        self._resolve_assistant_definition(assistant_def)
+        return assistant_def
+
+    def _resolve_assistant_definition(self, assistant_definition: AssistantDefinition):
+        assistant_definition.assistant_tools = {}
+        for tool in assistant_definition.tools:
+            if tool["type"] in ("code_interpreter", "retrieval"):
+                assistant_definition.assistant_tools[tool["type"]] = AssistantTool(
+                    name=tool["type"], openai_definition=tool, func=None
+                )
+            elif tool["type"] == "function":
+                function_tool = self._load_tool_as_function(tool)
+                assistant_definition.assistant_tools[function_tool.name] = function_tool
+            else:
+                raise UnsupportedAssistantToolType(
+                    message_format="Unsupported assistant tool type: {tool_type}",
+                    tool_type=tool["type"],
+                    target=ErrorTarget.EXECUTOR,
+                )
+
+    def _load_tool_as_function(self, tool: dict):
+        node, predefined_inputs = self._create_node_for_assistant_tool(tool)
+        resolved_tool = self.resolve_tool_by_node(node, convert_input_types=True)
+        func_name = resolved_tool.definition.function
+        definition = self._generate_tool_definition(func_name, resolved_tool.definition.description, predefined_inputs)
+        if resolved_tool.node.inputs:
+            inputs = {name: value.value for name, value in resolved_tool.node.inputs.items()}
+            func = partial(resolved_tool.callable, **inputs)
+        else:
+            func = resolved_tool.callable
+        return AssistantTool(name=func_name, openai_definition=definition, func=func)
+
+    def _create_node_for_assistant_tool(self, tool: dict):
```

Comment:
Why should we rely on nodes concept for asssitant tool? I think it should be decoupled.

Reply:
Th reason is to reuse existing method for input resolve. Per offline discussion, we keep it as it is for now and refine this logic later